### ISSUE TITLE
Prepare imglib2-ij for modularization

### ISSUE THREAD

**Issue by tpietzsch** [imglib/imglib2-ij#37]
We'll start by fixing imglib2-ij.

- [ ] Set `<package-name>net.imglib2.ij</package-name>`.
- [ ] Move all classes under the `net.imglib2.ij` package.

To update dependant projects, we need tooling to
- [ ] Find and fix import statements that reference imglib2-ij classes.
- [ ] Find and fix fully qualified uses of imglib2-ij classes.
- [ ] Create PRs to the dependant projects.

### Find and fix import statements that reference imglib2-ij classes.

- [ ] Clone all relevant candidate repos.
- [ ] List types (and interfaces, and enums) of imglib2-ij.

**Comment by acardona** [comment]
Changing the package name will break all existing documentation and
tutorials online. Is this necessary?

**Comment by tpietzsch** [comment]
Yes, it is necessary. JPMS requires that there are no split packages between artefacts.
(The same is true for OSGi. If I remember correctly, KNIME image processing for example had to work around this by having their own version of imglib2-ij.)

Breaking documentation and examples is a very good point. I did not think of that. We should also have scripts to update these. However its more difficult to gather all relevant repositories. (For code, we can at least harvest everything managed by pom-scijava.) Feel free to start putting inks in the comments of this issue!

**Comment by acardona** [comment]
Will also brake all user scripts. Most aren't under version control. Might
want to make this a breaking change with a separate Fiji version, leaving
the old untouched and not receiving more updates, to avoid chaos and
confusion.

**Comment by tpietzsch** [comment]
Another good point.

Maybe another solution might be to have a new artefact (`imglib2-ij2` or similar) with the new package name and also leave the old `imglib2-ij.jar` with deprecated classes. This would keep existing scripts working, and postpone the separate Fiji version until all of Fiji wants to run modularized. I would still want to move as much code and documentation as possible to the new package name. What do you think about that?

I would be also very interested in @ctrueden thoughts on this.

**Comment by gselzer** [comment]
@tpietzsch has there been any further progress on this? Would be great to utilize this work in SciJava Ops, but we are stuck here (see the [branch](https://github.com/scijava/scijava/tree/more-image-converters)). If there is any way I can help, let me know!

> Maybe another solution might be to have a new artefact (`imglib2-ij2` or similar) with the new package name and also leave the old `imglib2-ij.jar` with deprecated classes.

I like this idea!

**Comment by ctrueden** [comment]
> `imglib2-ij2`

How about `imglib2-imagej`? It is an integration layer between ImgLib2 and the original ImageJ API, after all. The name `imglib2-ij2` might rather imply an integration layer between ImgLib2 and ImageJ2, which would not be accurate.

The package prefix could be `net.imglib2.imagej` or `net.imglib2.ij`, whichever people like more.

**Comment by tpietzsch** [comment]
@gselzer Unfortunately there has been no progress on this, and it's low on my list of priorities currently.

If you want to work on it, you could just make `imglib2-imagej` (as @ctrueden suggested), and copy everything from `imglib2-ij` there under the `net.imglib2.imagej` package, github-actionify, make a beta release. Maybe start digging through the code reduce class and method visibility as much as possible (while we have the chance). Maybe start migrating some dependant projects to inform what needs to be `public`.

**Comment by ctrueden** [comment]
The imglib2-imagej component [now exists](https://github.com/imglib/imglib2-imagej). See [its issue tracker](https://github.com/imglib/imglib2-imagej/issues) for next steps and progress.

**Comment by odinsbane** [comment]
What about creating a tool to migrate package names for scripts? When the java 21 version comes out, include a validate script type function and it can be used to find any old legacy package names?
